### PR TITLE
fossil: Bump to 2.16

### PR DESCRIPTION
**Pull request by thunder-coding**

### CODE REVIEW

**Review by vishnuaq — APPROVED** [review]
ping
packages/poppler/build.sh

**Comment by thunder-coding** [comment]
Isn't this ready to merge?

File: packages/fossil/src-main.c.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
+--- ./src/main.c	2021-07-02 18:16:01.000000000 +0530
++++ ./src/main.c.mod	2021-08-03 07:01:36.763978266 +0530
+@@ -351,11 +351,11 @@
+   */
+   db_unsave_encryption_key();
+ #endif
+-#if defined(_WIN32) || (defined(__BIONIC__) && !defined(FOSSIL_HAVE_GETPASS))
```
Comment (Grimler91) [inline packages/fossil/src-main.c.patch:7]: Looks like the configure script checks for termux specifically and defines FOSSIL_HAVE_GETPASS if found, but I guess the check is broken with the autosetup-cc.tcl.patch

File: packages/fossil/autosetup-cc.tcl.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
+--- ./autosetup/cc.tcl	2021-07-02 18:16:01.000000000 +0530
++++ ./autosetup/cc.tcl.mod	2021-08-03 07:04:16.933978205 +0530
+@@ -109,25 +109,7 @@
+ #
+ # Checks that the given include files can be used.
+ proc cc-check-includes {args} {
+-	cc-check-some-feature $args {
```
Comment (Grimler91) [inline packages/fossil/autosetup-cc.tcl.patch:7]: Why did you need to disable this?

**Review by Grimler91 — COMMENTED** [review]
It looks like it should be possible to use getpass from [unistd.h](https://github.com/termux/termux-packages/blob/54975d7db5d4e2442f983158c9b51745e400b27f/ndk-patches/unistd.h.patch), why did that not work in 2.16?

File: packages/fossil/autosetup-cc.tcl.patch (hunk 2 above)
Comment (thunder-coding) [reply packages/fossil/autosetup-cc.tcl.patch:7]: The checks are probably broken. They don't work on device builds. I didn't test in the CI. Trying it now

File: packages/fossil/src-main.c.patch (hunk 1 above)
Comment (thunder-coding) [reply packages/fossil/src-main.c.patch:7]: getpass is available in unistd.h but not freepass

**Comment by Grimler91** [comment]
Please rebase against master so that it only builds fossil in CI
